pkg/webhooks/subscription/subscription.go

```diff
@@ -0,0 +1,183 @@
+}
+
+// Rules implements Webhook interface
+func (s *SubscriptionWebhook) Rules() []admissionregv1.RuleWithOperations { return rules }
+
+// GetURI implements Webhook interface
+func (s *SubscriptionWebhook) GetURI() string { return "/subscription-validation" }
+
+// SideEffects implements Webhook interface
+func (s *SubscriptionWebhook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// Validate is the incoming request even valid?
+func (s *SubscriptionWebhook) Validate(req admissionctl.Request) bool {
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+	valid = valid && (req.Kind.Kind == "Subscription")
+
+	return valid
+}
+
+// isLogging45Request is a helper to consolidate logic. Returns true
+// if the request is for the cluster-logging or elasticsearch-operator
+// subscription on 4.5 channel
+func (s *SubscriptionWebhook) isLogging45Request(subscriptionReq *subscriptionRequest) bool {
+	if subscriptionReq.Spec.Channel == BlockedChannel && (subscriptionReq.Spec.Name == LoggingSubName || subscriptionReq.Spec.Name == ESSubName) {
+		return true
+	}
+	return false
+}
+
+func (s *SubscriptionWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+	var err error
+	subReq := &subscriptionRequest{}
+
+	// if we delete, then look to OldObject in the request.
+	if request.Operation == v1beta1.Delete {
+		err = json.Unmarshal(request.OldObject.Raw, subReq)
+	} else {
+		err = json.Unmarshal(request.Object.Raw, subReq)
+	}
+
+	if err != nil {
+		ret = admissionctl.Errored(http.StatusBadRequest, err)
+		return ret
+	}
+
+	// can comment this out or remove after manual tests
+	log.Info("User is attempting to modify subscription", "username", request.AdmissionRequest.UserInfo.Username, "operation", request.Operation, "subscription name", subReq.Spec.Name, "channel", subReq.Spec.Channel)
+
+	// If this isn't a request to install or upgrade logging 4.5, let RBAC handle this
+	if !s.isLogging45Request(subReq) {
+		ret = admissionctl.Allowed("Base decisions for non-logging subscriptions on RBAC")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+
+	// Admin users
+	if utils.SliceContains(request.AdmissionRequest.UserInfo.Username, privilegedUsers) {
+		ret = admissionctl.Allowed("Admin users are may install or upgrade logging 4.5 operator")
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+	// Users in admin groups
+	for _, group := range request.AdmissionRequest.UserInfo.Groups {
+		if utils.SliceContains(group, adminGroups) {
+			ret = admissionctl.Allowed("Members of admin group may install or upgrade logging 4.5 operator")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	// if we're here, non-privileged user is attempting to CREATE or UPDATE logging
+	// operator at 4.5 - deny this
+	ret = admissionctl.Denied("Non-privileged users cannot install or upgrade logging 4.5 operator")
```

Comment:
Do we want to be more straight forward here, stating there are known issues with the logging-operator v4.5?

Reply:
yes. I'll add in more info here

Reply:
something like ""Non-privileged users cannot install or upgrade to the v4.5 logging operator at this time, as there are known issues with logging v4.5 which we are working to resolve." ?